File: app/src/lib/backend/github.ts
import { invoke } from '$lib/backend/ipc';

export type Verification = {
	user_code: string;
	device_code: string;
};

export async function initDeviceOauth() {
	return invoke<Verification>('init_device_oauth');
}

export async function checkAuthStatus(params: { deviceCode: string }) {
	return invoke<string>('check_auth_status', params);
}

File: .conflict-side-1/crates/gitbutler-tauri/src/github.rs
use but_api::{
    commands::github::{self, Verification},
    App, NoParams,
};
use tauri::State;
use tracing::instrument;

use but_api::error::Error;

#[tauri::command(async)]
#[instrument(skip(app), err(Debug))]
pub async fn init_device_oauth(app: State<'_, App>) -> Result<Verification, Error> {
    github::init_device_oauth(&app, NoParams {}).await
}

#[tauri::command(async)]
#[instrument(skip(app), err(Debug))]
pub async fn check_auth_status(app: State<'_, App>, device_code: String) -> Result<String, Error> {
    github::check_auth_status(&app, github::CheckAuthStatusParams { device_code }).await
}
